spring-rabbit/src/main/java/org/springframework/amqp/rabbit/connection/ConnectionFactoryUtils.java

```diff
@@ -72,6 +72,22 @@ public static boolean isChannelTransactional(Channel channel, ConnectionFactory
 	 */
 	public static RabbitResourceHolder getTransactionalResourceHolder(final ConnectionFactory connectionFactory,
 			final boolean synchedLocalTransactionAllowed) {
+		return getTransactionalResourceHolder(connectionFactory, synchedLocalTransactionAllowed, false);
+	}
+
+	/**
+	 * Obtain a RabbitMQ Channel that is synchronized with the current transaction, if any.
+	 * @param connectionFactory the ConnectionFactory to obtain a Channel for
+	 * @param synchedLocalTransactionAllowed whether to allow for a local RabbitMQ transaction that is synchronized with
+	 * a Spring-managed transaction (where the main transaction might be a JDBC-based one for a specific DataSource, for
+	 * example), with the RabbitMQ transaction committing right after the main transaction. If not allowed, the given
+	 * ConnectionFactory needs to handle transaction enlistment underneath the covers.
+	 * @param publisherConnectionIfPossible obtain a connection from a separate publisher connection
+	 * if possible.
+	 * @return the transactional Channel, or <code>null</code> if none found
+	 */
+	public static RabbitResourceHolder getTransactionalResourceHolder(final ConnectionFactory connectionFactory,
```

Comment:
Why this code is still here?
I thought the `getPublisherConnectionFactory()` would be enough to distinct the CF instance inside the `RabbitTemplate`.

Reply:
No; we can't do that - the connection factory might be a `RoutingConnectionFactory` so there is not one publishing CF.

Reply:
Still unclear how this transactional stuff is related to the `RoutingConnectionFactory`.
In the end you perform:
```
if (publisherConnectionIfPossible) {
			ConnectionFactory publisherFactory = connectionFactory.getPublisherConnectionFactory();
```
But if we select the `publsherCF` before calling this `getTransactionalResourceHolder()`, we don't need the logic around `publisherConnectionIfPossible` here at all. Isn't it?